minstack.py

```diff
@@ -0,0 +1,47 @@
+class MinStack(object):
+
+    def __init__(self):
+        self.s = []
+
+    def push(self,x):
+        cur_min = self.getMin()
+        if cur_min == None or x < cur_min:
+            cur_min = x
+        self.s.append((x,cur_min))
+
+    def pop(self):
```

Comment:
should you pop by checking some of the conditions or you should only pop the topmost element ?